sycl/test-e2e/Matrix/get_coordinate_ops_impl.hpp

```diff
@@ -0,0 +1,199 @@
+                                sycl::memory_scope::device>
+                   aref(v_cols[i]);
+               aref.fetch_add(sum_local_cols[i]);
+             }
+           }
+         }); // parallel for
+   }).wait();
+}
+
+template <typename T, typename TResult, size_t SROWS, size_t SCOLS, use Use,
+          layout Layout, size_t VF>
+void test_get_coord_op() {
+  constexpr size_t SCALE = 2;
+  static constexpr size_t Rows = SROWS * SCALE;
+  static constexpr size_t Cols = SCOLS * SCALE;
+
+  T M[Rows][Cols];
+  T Mvnni[Rows / VF][Cols * VF];
+  TResult sum_rows[Rows] = {0};
+  TResult sum_rows_ref[Rows] = {0};
+  TResult sum_cols[Cols] = {0};
+  TResult sum_cols_ref[Cols] = {0};
+
+  for (int i = 0; i < Rows; i++) {
+    for (int j = 0; j < Cols; j++) {
+      M[i][j] = i + j;
+    }
+  }
+
+  matrix_vnni<T>(Rows, Cols, *M, *Mvnni, VF);
+  big_matrix<T, Rows / VF, Cols * VF> MM((T *)&Mvnni);
+
+  matrix_sum<T, TResult, Rows, Cols, SROWS, SCOLS, Use, Layout, VF>(
+      MM, sum_rows, sum_cols);
+
+  for (int i = 0; i < Rows; i++) {
+    for (int j = 0; j < Cols; j++) {
+      sum_rows_ref[i] += (int)M[i][j];
+    }
+    assert(std::fabs(sum_rows_ref[i] - sum_rows[i]) <= FLOAT_EPSILON);
+  }
+
+  for (int j = 0; j < Cols; j++) {
+    for (int i = 0; i < Rows; i++) {
+      sum_cols_ref[j] += (int)M[i][j];
+    }
+    assert(std::fabs(sum_cols_ref[j] - sum_cols[j]) <= FLOAT_EPSILON);
+  }
+}
+
+int main() {
+  queue q;
+  std::vector<combination> combinations =
+      q.get_device()
+          .get_info<sycl::ext::oneapi::experimental::info::device::
+                        matrix_combinations>();
+
+  for (unsigned int i = 0; i < combinations.size(); i++) {
+    if (combinations[i].nsize == 0) { // Intel AMX
+      test_get_coord_op<int8_t, int, /*TM*/ 16, /*TK*/ 64, use::a,
+                        layout::row_major, 1>();
+      test_get_coord_op<int8_t, int32_t, /*TK*/ 64, /*TN*/ 16, use::b,
+                        layout::ext_intel_packed, 4>();
+      test_get_coord_op<float, float, /*TM*/ 16, /*TN*/ 16, use::accumulator,
+                        layout::row_major, 1>();
+      break;
+    }
+
+    if (combinations[i].nsize == 16) { // architecture::intel_gpu_pvc
+      test_get_coord_op<bfloat16, float, /*TM*/ 8, /*TK*/ 16, use::a,
+                        layout::row_major, 1>();
+      test_get_coord_op<int8_t, int, /*TM*/ 8, /*TK*/ 32, use::a,
+                        layout::row_major, 1>();
+      test_get_coord_op<bfloat16, float, /*TK*/ 16, /*TN*/ 16, use::b,
+                        layout::row_major, 1>();
+      test_get_coord_op<bfloat16, float, /*TK*/ 16, /*TN*/ 16, use::b,
+                        layout::ext_intel_packed, 2>();
+      test_get_coord_op<int8_t, int32_t, /*TK*/ 32, /*TN*/ 16, use::b,
+                        layout::ext_intel_packed, 4>();
+      test_get_coord_op<float, float, /*TM*/ 8, /*TN*/ 16, use::accumulator,
+                        layout::row_major, 1>();
```

Comment:
Awesome, please add the extra cases to DG2 and AMX as well: 
- a row major bfloat16
- b row major bfloat16
- b packed bfloat16
- b row major int8
- c int32 row major 

Also, I see that - b row major int8 is missing in PVC. Please add that as well

Reply:
Thanks for the suggestion! I have included each of the following cases for DG2, AMX  and PVC. 
- a row_major  bloat16 
- a row_major  int8
- b row_major bloat16
- b row_major int8
- b packed bloat16
- b packed int8
- a float
- a int32

Please feel free to let me know if there is any other cases need to be added :)